pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotInstanceAssignmentRestletResource.java

```diff
@@ -244,20 +244,32 @@ public Map<String, InstancePartitions> assignInstances(
   private void assignInstancesForInstancePartitionsType(Map<String, InstancePartitions> instancePartitionsMap,
       TableConfig tableConfig, List<InstanceConfig> instanceConfigs, InstancePartitionsType instancePartitionsType) {
     String tableNameWithType = tableConfig.getTableName();
-    if (TableConfigUtils.hasPreConfiguredInstancePartitions(tableConfig, instancePartitionsType)) {
+    if (!TableConfigUtils.hasPreConfiguredInstancePartitions(tableConfig, instancePartitionsType)) {
```

Comment:
If a tableConfig doesn't have the InstancePartitionMap configured, we will always come into this if block. 
That means we'll never reach the "else if" block on L254 if instancePartitionMap is not configured. 
Is that the right behavior? 

My understanding was that we can use PRE_CONFIGURATION_INSTANCE_SELECTOR as long as InstanceAssignmentConfigMap in tableConfig is set.

Reply:
It's inteneded. The new PRE_CONFIGURATION_INSTANCE_SELECTOR needs a reference instance parition map (preconfigured) to work, therefore if instancePartitionsMap is missing we cannot use PRE_CONFIGURATION_INSTANCE_SELECTOR.